README.md
# Polybase-server
WIP

## Setup

### Requirements

- Ubuntu 20.04
- SSH root

### Install dependencies

```sh
curl https://raw.githubusercontent.com/mardix/polybase-server/main/setup.sh > polybase-server-setup.sh
chmod 755 polybase-server-setup.sh
./polybase-server-setup.sh
```

---

License: MIT


CHANGELOG
# Polybase Server


## Version: 1.0.0
- ARANGODB_VERSION=3.7.9-1
- TYPESENSE_VERSION=0.19.0
- INFLUXDB_VERSION=1.8
- REDIS_VERSION=latest

backup.sh
# --------------------------------------------------------------------------
# Polybase-Server backup.sh
#
# curl https://raw.githubusercontent.com/mardix/lunabase-server/master/backup.sh > lunbase-server-backup.sh
# chmod 755 lunabase-server-backup.sh
# ./lunabase-server-backup.sh
#--------------------------------------------------------------------------
# *** Example with incremental backup
# > ./lunabase-server-backup.sh
# *** To archive a backup, add -a. It will place the backup inside of /archive
# > ./lunabase-server-backup.sh --archive
################################################################
################################################################
# CONFIGURATION
# AWS
CONF_AWS_BUCKET_NAME="BUCKET"

# 
# DATA_LIST 
# List of data dir to backup to S3
# format: $NAME:$DATA_DIR
# one data dir per line
#
DATA_LIST="
redis:/var/lib/redis
arango:/var/lib/arangodb3
typesense:/var/lib/typesense
influxdb:/var/lib/influxdb/data
"

# --------------------------------------------------------------
################################################################
# STOP!!!
# DO NOT MODIFY BELOW, UNLESS YOU KNOW WHAT YOU ARE DOING :) 
################################################################
APP_NAME="Polybase-Server"
# Backup time
DT_NOW=`date +%H00`
# Datetime for archive backup
DT_ARCHIVE=`date +%Y%m%d%H00`
# Daily name, from 0=sunday, 6=saturday -> daily.0
DT_DAILY=$(date +"%H00")



echo 
echo "$APP_NAME backup..."
echo 

for fname in $DATA_LIST
do
  IFS=":" read -a data <<< "$fname"
  NAME=${data[0]}
  DATA_DIR=${data[1]}
  BACKUP_NAME=$NAME
  BACKUP_FILENAME=$BACKUP_NAME.tar.gz
  BACKUP_FILE=/tmp/backup_$BACKUP_FILENAME
  BACKUP_OBJECT=daily/$DT_DAILY/$BACKUP_FILENAME
  
  # archive
  if [ "$1" ]; then
      BACKUP_OBJECT=archive/$DT_ARCHIVE/$BACKUP_FILENAME
  fi
  
  echo "> $BACKUP_NAME"
  echo "   - starting: $(date +"%Y-%m-%d %H:%M:%S") ..."
  echo "   - object name: $BACKUP_OBJECT"
  echo "   - archiving $DATA_DIR -> $BACKUP_FILE ..."
  #tar -zcf $BACKUP_FILE $DATA_DIR
  echo "   - uploading to AWS S3 $CONF_AWS_BUCKET_NAME/$BACKUP_OBJECT ..."
  #aws put --no-vhost $CONF_AWS_BUCKET_NAME/$BACKUP_OBJECT $BACKUP_FILE
  echo "   - removing temp file $BACKUP_FILE ..."  
  #rm -rf $BACKUP_FILE
  echo "   - completed: $(date +"%Y-%m-%d %H:%M:%S") ..."
  echo 
done
echo "$APP_NAME backup completed!"
echo 

setup copy.sh
# --------------------------------------------------------------------------
# Polybase Server setup
#
# curl https://raw.githubusercontent.com/mardix/polybase-server/master/setup.sh > polybase-server-setup.sh
# chmod 755 polybase-server-setup.sh
# ./polybase-server-setup.sh
#--------------------------------------------------------------------------
#--------------------------------------------------------------------------
#--------------------------------------------------------------------------

# CONFIGURATION
# Configure packages version in case 
ARANGODB_VERSION=3.7.9-1
TYPESENSE_VERSION=0.19.0

#--------------------------------------------------------------------------
echo "------------------------------------------------"
echo "::Polybase server setup::"
echo

apt-get update
apt-get install -y wget curl cron build-essential
apt-get update

# Install Polybox
echo 
echo "=======>>> installing Polybox ..."
echo 
curl https://raw.githubusercontent.com/mardix/polybox/master/install.sh > polybox-install.sh
chmod 755 polybox-install.sh
./polybox-install.sh

# Install Redis
echo 
echo "------------------------------------------------"
echo "=======>>> installing Redis ..."
echo 
while true; do
    read -s -p "Enter Redis Password: " password
    echo
    read -s -p "Re-enter Redis Password: " password2
    echo
    [ "$password" = "$password2" ] && break
    echo "Passwords didn't match. Try again"
done
echo
apt-get install -y redis-server
sed -i 's/bind 127.0.0.1/bind 0.0.0.0/g' /etc/redis/redis.conf
sed -i 's/supervised.*/supervised systemd/g' /etc/redis/redis.conf
sed -i "s/# requirepass.*/requirepass $password/g" /etc/redis/redis.conf
sudo systemctl restart redis.service


# Install ArangoDB
echo 
echo "------------------------------------------------"
echo "=======>>> installing ArangoDB "
echo 
curl -OL https://download.arangodb.com/arangodb37/DEBIAN/Release.key
sudo apt-key add - < Release.key
echo 'deb https://download.arangodb.com/arangodb37/DEBIAN/ /' | sudo tee /etc/apt/sources.list.d/arangodb.list
sudo apt-get install apt-transport-https
sudo apt-get update
sudo apt-get install arangodb3=$ARANGODB_VERSION
sudo systemctl start arangodb3
sudo systemctl enable arangodb3
sed -i 's/endpoint = tcp:\/\/127.0.0.1:8529/endpoint = tcp:\/\/0.0.0.0:8529/g' /etc/arangodb3/arangod.conf
sudo systemctl restart arangodb3

# Install Typesense
echo 
echo "------------------------------------------------"
echo "=======>>> installing Typesense "
echo 
curl https://dl.typesense.org/releases/$TYPESENSE_VERSION/typesense-server-$TYPESENSE_VERSION-amd64.deb -o typesense-server.deb
apt install -y ./typesense-server.deb
sudo systemctl enable typesense-server
sudo systemctl restart typesense-server

# Setup backup
echo
echo 
echo "------------------------------------------------"
echo "=======>>> setting up backup system "
echo 
echo " - setting awscli..."
curl https://raw.githubusercontent.com/timkay/aws/master/aws -o aws
chmod +x aws
mv aws /usr/bin
touch ~/.awssecret
chmod go-rwx ~/.awssecret
echo
echo " - setting backup.sh..."
curl https://raw.githubusercontent.com/mardix/polybase-server/master/backup.sh > /etc/polybase-server-backup.sh
chmod 755 /etc/polybase-server-backup.sh
echo
echo " - setting cron file..."
rm -rf /etc/cron.d/polybase-server
curl https://raw.githubusercontent.com/mardix/polybase-server/master/cron > /etc/cron.d/polybase-server
echo
echo

echo "::Polybase server setup completed::"
echo
echo "------------------------------------------------"
# EOF

setup.sh
# --------------------------------------------------------------------------
# Polybase Server setup
#
# curl https://raw.githubusercontent.com/mardix/polybase-server/main/setup.sh > polybase-server-setup.sh
# chmod 755 polybase-server-setup.sh
# ./polybase-server-setup.sh
#--------------------------------------------------------------------------
#--------------------------------------------------------------------------
#--------------------------------------------------------------------------

# CONFIGURATION
POLYBASE_VERSION=1.0.0

# DEPENDENCIES VERSION
ARANGODB_VERSION=3.7.9-1
TYPESENSE_VERSION=0.19.0
INFLUXDB_VERSION=1.8
REDIS_VERSION=latest

#--------------------------------------------------------------------------
echo "------------------------------------------------"
echo "::Polybase server setup::"
echo
echo

# Main password
while true; do
    read -s -p "Enter Main Server Password: " MAINPWD
    echo
    read -s -p "Re-enter Main Server Password: " MAINPWD2
    echo
    [ "$MAINPWD" = "$MAINPWD2" ] && break
    echo "Passwords didn't match. Try again"
done
echo
if [ -z "$MAINPWD" ]; then
    echo
    echo "ERROR: Password can't (and shouldn't...) be empty! Please provide a valid value!"
    exit 1
fi

# Update system
echo 
echo "=======>>> update system ..."
echo 
apt-get update
apt-get install -y wget curl cron build-essential
apt-get update

# Install Polybox
echo 
echo "=======>>> installing Polybox ..."
echo 
curl https://raw.githubusercontent.com/mardix/polybox/master/install.sh > polybox-install.sh
chmod 755 polybox-install.sh
./polybox-install.sh

# Install Redis
echo 
echo "------------------------------------------------"
echo "=======>>> installing Redis ..."
apt-get install -y redis-server
sed -i 's/bind 127.0.0.1/bind 0.0.0.0/g' /etc/redis/redis.conf
sed -i 's/supervised.*/supervised systemd/g' /etc/redis/redis.conf
sed -i "s/# requirepass.*/requirepass $MAINPWD/g" /etc/redis/redis.conf
sudo systemctl restart redis.service


# Install ArangoDB
echo 
echo "------------------------------------------------"
echo "=======>>> installing ArangoDB "
echo 
curl -OL https://download.arangodb.com/arangodb37/DEBIAN/Release.key
sudo apt-key add - < Release.key
echo 'deb https://download.arangodb.com/arangodb37/DEBIAN/ /' | sudo tee /etc/apt/sources.list.d/arangodb.list
sudo apt-get install apt-transport-https
sudo apt-get update
echo arangodb3 arangodb3/password password "$MAINPWD" | debconf-set-selections
echo arangodb3 arangodb3/password_again password "$MAINPWD" | debconf-set-selections
sudo DEBIAN_FRONTEND=noninteractive apt-get install -y -q arangodb3=$ARANGODB_VERSION
sed -i 's/endpoint = tcp:\/\/127.0.0.1:8529/endpoint = tcp:\/\/0.0.0.0:8529/g' /etc/arangodb3/arangod.conf
sudo systemctl start arangodb3
sudo systemctl enable arangodb3
sudo systemctl restart arangodb3

# Install Typesense
echo 
echo "------------------------------------------------"
echo "=======>>> installing Typesense "
echo 
curl https://dl.typesense.org/releases/$TYPESENSE_VERSION/typesense-server-$TYPESENSE_VERSION-amd64.deb -o typesense-server.deb
sudo apt install ./typesense-server.deb
sudo systemctl enable typesense-server
sudo systemctl restart typesense-server

# Install InfluxDB 1.8
echo 
echo "------------------------------------------------"
echo "=======>>> installing Influxdb "
echo 
curl -s https://repos.influxdata.com/influxdb.key | sudo apt-key add -
source /etc/lsb-release
echo "deb https://repos.influxdata.com/${DISTRIB_ID,,} ${DISTRIB_CODENAME} stable" | sudo tee /etc/apt/sources.list.d/influxdb.list
sudo apt-get update && sudo apt-get install influxdb
sudo systemctl unmask influxdb.service
sudo systemctl start influxdb
influx -execute "CREATE USER root WITH PASSWORD '$MAINPWD' WITH ALL PRIVILEGES"
sudo systemctl restart influxdb
sed -i 's/# auth-enabled = false/auth-enabled = true/g' /etc/influxdb/influxdb.conf
sed -i 's/# pprof-auth-enabled = false/pprof-auth-enabled = true/g' /etc/influxdb/influxdb.conf
sed -i 's/# ping-auth-enabled = false/ping-auth-enabled = true/g' /etc/influxdb/influxdb.conf
sudo systemctl restart influxdb

echo "::Polybase server setup completed::"
echo
echo "------------------------------------------------"
# EOF

